src/layout/layout.rs

```diff
@@ -41,6 +41,46 @@ thread_local! {
     ));
 }
 
+#[derive(Debug, Clone, Eq, PartialEq, Hash)]
+pub enum Spacing {
+    Space(u16),
+    Overlap(u16),
+}
+
+impl Default for Spacing {
+    fn default() -> Self {
+        Self::Space(0)
+    }
+}
+
+impl From<u32> for Spacing {
+    fn from(value: u32) -> Self {
+        Self::from(value.clamp(u32::from(u16::MIN), u32::from(u16::MAX)) as u16)
+    }
```

Comment:
Why do we need from u32? There shouldn't be any code that already uses u32 for backwards compat right?

Reply:
I thought I tried that first. If I don't implement `i32`, I get compile errors on some tests:

<img width="1359" alt="image" src="https://github.com/user-attachments/assets/29da24de-62d1-4417-919e-0fd2b58d7927">

But it looks like implementing `u32` is not required. Resolved in https://github.com/ratatui/ratatui/pull/1398/commits/3b817e598a49c591ab51b032507a12a9808f1c85